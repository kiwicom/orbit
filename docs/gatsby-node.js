// creates a "collection" field to it easier to filter nodes created by
// gatsby-source-filesystem instead of awkwardly filtering by file path
// https://github.com/gatsbyjs/gatsby/issues/1634#issuecomment-388899348
exports.onCreateNode = ({ node, getNode, actions }) => {
  if (node.internal.type !== "Mdx") return;
  const { createNodeField } = actions;
  const parent = getNode(node.parent);
  createNodeField({
    node,
    name: "collection",
    value: parent.sourceInstanceName,
  });
};

exports.createPages = async ({ graphql, actions, reporter }) => {
  const { createPage } = actions;
  const result = await graphql(`
    query DocumentationQuery {
      allMdx(filter: { fields: { collection: { eq: "documentation" } } }) {
        nodes {
          frontmatter {
            title
          }
          body
          slug
        }
      }
    }
  `);
  if (result.errors) {
    reporter.panicOnBuild(`Error when querying guides.`);
    return;
  }
  result.data.allMdx.nodes.forEach(guide => {
    const slug = guide.slug
      .split("/")
      .map(s => s.replace(/^\d+-\s*/g, ""))
      .join("/");

    createPage({
      path: `/${slug}`,
      component: `${__dirname}/src/templates/Guide.tsx`,
      context: {
        title: guide.frontmatter.title,
        body: guide.body,
      },
    });
  });
};
