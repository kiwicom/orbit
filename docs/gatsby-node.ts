import path from "path";
import fs from "fs";
import yaml from "js-yaml";
import { createFilePath } from "gatsby-source-filesystem";
import type { GatsbyNode } from "gatsby";

import { omitNumbers, getDocumentUrl, getParentUrl, getDocumentTrail } from "./utils/document";
import getOverviewPages from "./services/overviewPages";
import parseChangelog from "./services/changelog";

const ROOT = path.resolve(__dirname, "..");

export const onCreateNode = async ({ cache, node, getNode, actions, reporter }) => {
  if (
    node.internal.type === "Directory" &&
    node.sourceInstanceName === "documentation" &&
    node.relativePath !== "" &&
    !node.relativePath.startsWith("assets")
  ) {
    const metaFilePath = path.join(node.absolutePath, "meta.yml");
    const url = omitNumbers(path.join("/", node.relativePath, "/"));

    if (!fs.existsSync(metaFilePath)) {
      reporter.panicOnBuild(
        `Expected meta.yml file to exist in "${
          process.env.NODE_ENV === "test"
            ? path.relative(ROOT, node.absolutePath)
            : node.absolutePath
        }", every directory in "src/documentation" should have one`,
      );
      await cache.set(url, {});
    }
  }

  /**
   * TODO: updating meta.yml files should update document navigation and breadcrumbs,
   * an obvious solution for this is to compute `trail` on frontend rather than here,
   * but we wanted to avoid large and repetitive GraphQL queries.
   *
   * A temporary and clumsy way to see meta.yml updates appears to be editing an MDX file
   * within the same directory.
   */

  if (
    node.internal.type === "File" &&
    node.base === "meta.yml" &&
    node.sourceInstanceName === "documentation"
  ) {
    const metaFilePath = node.absolutePath;
    const metaFilePathRelative = path.relative(ROOT, metaFilePath);
    const url = omitNumbers(path.join("/", node.relativeDirectory, "/"));
    const metaFileData = yaml.load(fs.readFileSync(metaFilePath));
    const missingFields = [];

    if (typeof metaFileData.title === "undefined") {
      // @ts-expect-error TODO
      missingFields.push("title");
    }
    if (typeof metaFileData.type === "undefined") {
      // @ts-expect-error TODO
      missingFields.push("type");
    } else if (!["folder", "tabs"].includes(metaFileData.type)) {
      reporter.panicOnBuild(
        `Expected the value of "type" in "${
          process.env.NODE_ENV === "test" ? metaFilePathRelative : metaFilePath
        }" to be one of: folder, tabs`,
      );
    }

    if (missingFields.length > 0) {
      reporter.panicOnBuild(
        `Expected "${
          process.env.NODE_ENV === "test" ? metaFilePathRelative : metaFilePath
        }" to contain fields: ${missingFields.join(", ")}`,
      );
    }

    await cache.set(url, metaFileData);
  }

  if (node.internal.type === "Mdx") {
    const { createNodeField } = actions;
    const parent = getNode(node.parent);
    const fileUrl = createFilePath({ node, getNode, basePath: `pages` });
    const { dir } = path.parse(fileUrl);
    const metaFileData = (await cache.get(getParentUrl(omitNumbers(fileUrl)))) || {};
    const hasTabs = metaFileData.type === "tabs";

    createNodeField({
      node,
      name: "slug",
      value: getDocumentUrl(fileUrl, hasTabs),
    });

    createNodeField({
      node,
      name: "description",
      value: node.frontmatter.description || node.description || metaFileData.description,
    });

    // creates a "collection" field to make it easier to filter nodes created by
    // gatsby-source-filesystem instead of awkwardly filtering by file path
    // https://github.com/gatsbyjs/gatsby/issues/1634#issuecomment-388899348

    createNodeField({
      node,
      name: "collection",
      value: parent.sourceInstanceName,
    });

    // Create a similar field that should be identical
    // for all pages that are tabs in one folder
    createNodeField({
      node,
      name: "tabCollection",
      value: hasTabs ? parent.relativeDirectory : null,
    });

    // Make title the same for all pages in tabs
    const title = hasTabs ? metaFileData.title : node.frontmatter.title;
    createNodeField({
      node,
      name: "title",
      value: hasTabs ? metaFileData.title : node.frontmatter.title,
    });

    const getLinkForComponents = () => {
      if (dir.startsWith("/03-components")) {
        if (dir.match(/primitive/)) {
          return `https://github.com/kiwicom/orbit/tree/master/packages/orbit-components/src/primitives/${title}`;
        }
        if (dir.match(/grid/)) {
          return `https://github.com/kiwicom/orbit/tree/master/packages/orbit-components/src/utils/Grid`;
        }
        return `https://github.com/kiwicom/orbit/tree/master/packages/orbit-components/src/${title}`;
      }
      return "";
    };

    createNodeField({
      node,
      name: "headerLink",
      value: getLinkForComponents() || node.frontmatter.headerLink || metaFileData.headerLink,
    });

    if (node.fields.collection === "documentation") {
      let documentTrail;

      if (hasTabs && path.basename(node.fileAbsolutePath).startsWith("01-")) {
        documentTrail = await getDocumentTrail(cache, node.fields.slug);
      } else {
        documentTrail = await getDocumentTrail(cache, getParentUrl(node.fields.slug));
        documentTrail.push({
          name: node.frontmatter.title,
          url: node.fields.slug,
        });
      }
      createNodeField({
        node,
        name: "trail",
        value: documentTrail,
      });
    } else {
      createNodeField({
        node,
        name: "trail",
        value: [],
      });
    }
  }
};

export const createPages: GatsbyNode["createPages"] = async ({
  graphql,
  actions,
  reporter,
  cache,
}) => {
  const { createPage } = actions;
  const result = await graphql(`
    query DocumentationQuery {
      allMdx(filter: { fields: { collection: { eq: "documentation" } } }) {
        nodes {
          id
          fields {
            slug
            tabCollection
          }
        }
      }
    }
  `);

  const overviewPages = await getOverviewPages();

  await Promise.all(
    overviewPages.map(async page => {
      const trail = await getDocumentTrail(cache, page.slug);
      const overviewPath = path.join(__dirname, "../../src/templates/Overview.tsx");
      createPage({
        path: page.slug,
        component:
          process.env.NODE_ENV === "test" ? path.relative(ROOT, overviewPath) : overviewPath,
        context: { ...page, trail },
      });
    }),
  );

  if (result.errors) {
    reporter.panicOnBuild(`Error when querying guides.`);
    return;
  }

  // @ts-expect-error TODO
  result.data.allMdx.nodes.forEach(({ id, fields }) => {
    createPage({
      path: fields.slug,
      component: `${process.cwd()}/src/templates/Doc.tsx`,
      context: { id, tabs: fields.tabCollection },
    });
  });
};

export const onPreBuild = async () => {
  await parseChangelog();
};

export const createSchemaCustomization: GatsbyNode["createSchemaCustomization"] = ({ actions }) => {
  actions.createTypes(
    `
    type Mdx implements Node {
      frontmatter: MdxFrontmatter!
      fields: MdxFields
    }

    type MdxFrontmatter {
      date: Date @dateformat
      description: String
      redirect_from: [String]
      title: String!
      type: String
    }

    type MdxFields {
      collection: String!
      description: String
      slug: String!
      tabCollection: String
      title: String!
      trail: [TrailPart]
    }

    type TrailPart {
      name: String!
      url: String!
    }
  `,
  );
};
