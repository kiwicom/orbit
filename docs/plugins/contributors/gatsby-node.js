const { Octokit } = require("@octokit/rest");
const path = require("path");

const NODE = `contributor`;

exports.sourceNodes = async ({ actions, createContentDigest, createNodeId }, { repo, owner }) => {
  const { createNode, createTypes } = actions;

  createTypes(
    `
    type Contributor implements Node {
      id: ID!
      login: String
      avatar_url: String
      url: String
      name: String
      blog: String
      bio: String
      email: String
      location: String
      twitter_username: String
      company: String
    }
  `,
  );

  try {
    require("dotenv-safe").config({
      example: path.resolve(__dirname, `../../../.env.example`),
      path: path.resolve(__dirname, `../../../.env`),
      allowEmptyValues: true,
    });

    const octokit = new Octokit({ auth: process.env.GH_TOKEN });
    const names = [];

    await octokit.repos
      .listContributors({
        owner,
        repo,
      })
      .then(c => names.push(...c.data.map(({ login }) => login)));

    const reqs = names.map(username => octokit.users.getByUsername({ username }));
    const users = await Promise.all(reqs).then(all => all.map(n => n.data));

    users.forEach(u => {
      createNode({
        ...u,
        id: createNodeId(`${NODE}-${u.id}`),
        parent: null,
        children: [],
        internal: {
          type: NODE,
          content: JSON.stringify(u),
          contentDigest: createContentDigest(u),
        },
      });
    });
  } catch (error) {
    if (error.missing.includes("GH_TOKEN")) {
      console.warn("Missing an access token for GitHub. Please create one.");
      console.info(
        "The token is needed to display the Contributors component, which is based on the GitHub API.",
        "Create a personal access token: https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token",
        "Don't forget to include the required scopes for the token: all repo scopes and read:org.",
        "After creating a token in GitHub, add it to a .env file in the root folder with GH_TOKEN=<YOUR TOKEN>",
      );
    } else {
      console.error(error);
      console.info(
        "You may have forgotten to include the repo and read:org scopes for your GitHub access token.",
      );
    }
  }

  return [];
};
