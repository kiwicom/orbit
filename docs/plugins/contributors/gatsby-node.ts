import path from "path";
import fs from "fs-extra";
import type { GatsbyNode } from "gatsby";

const NODE = `contributor`;

// only reads static file generated by cron job and creates nodes

export const sourceNodes = async ({ actions, createContentDigest, createNodeId }) => {
  const { createNode } = actions;
  const staticData = JSON.parse(await fs.readFile(path.join(__dirname, "./users.json"), "utf8"));

  try {
    const users = await fs.readFile(path.join(__dirname, "fetchedUsers.json"), "utf8");

    staticData.concat(JSON.parse(users)).forEach(u => {
      return createNode({
        error: u.error || "",
        avatar_url: u.avatar_url || "",
        ...u,
        core: u.core || false,
        username: u.username || u.id,
        id: createNodeId(u.id),
        parent: null,
        children: [],
        internal: {
          type: NODE,
          content: JSON.stringify(u),
          contentDigest: createContentDigest(u),
        },
      });
    });
  } catch (error) {
    console.error(error);
    console.info(
      "You may have forgotten to include the repo and read:org scopes for your GitHub access token.",
    );
  }
};

export const createSchemaCustomization: GatsbyNode["createSchemaCustomization"] = ({ actions }) => {
  const { createTypes } = actions;

  const typeDefs = `
    type Contributor implements Node {
      id: ID!
      name: String
      info: String
      error: String
      username: String
      position: String
      dribbble: String
      github: String
      avatar_url: String
      twitter: String
      website: String
      active: Boolean
      core: Boolean
    }
  `;

  createTypes(typeDefs);
};
