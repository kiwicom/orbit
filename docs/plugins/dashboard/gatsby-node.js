exports.createSchemaCustomization = require("./gatsby/create-schema-customization");
exports.onPreInit = require("./gatsby/onPreInit");
