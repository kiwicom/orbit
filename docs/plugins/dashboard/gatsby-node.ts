import createSchemaCustomization from "./gatsby/createSchemeCustomization";
// import createPages from "./gatsby/createPages";

export { createSchemaCustomization };
