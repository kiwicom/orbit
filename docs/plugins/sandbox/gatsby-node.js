exports.createPages = require("./gatsby/on-create-page");
exports.sourceNodes = require("./gatsby/on-source-nodes");
exports.onCreateNode = require("./gatsby/on-create-node");
exports.createSchemaCustomization = require("./gatsby/create-schema-customization");

exports.onCreateWebpackConfig = ({ actions }) => {
  actions.setWebpackConfig({
    node: {
      fs: "empty",
    },
  });
};
