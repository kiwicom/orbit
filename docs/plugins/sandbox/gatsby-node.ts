import sourceNodes from "./gatsby/sourceNodes";
import createSchemaCustomization from "./gatsby/createSchemeCustomization";

export { sourceNodes, createSchemaCustomization };
