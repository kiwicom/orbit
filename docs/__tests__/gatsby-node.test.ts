/**
 * @jest-environment node
 */
import path from "path";
import { vol } from "memfs";
import dedent from "dedent";
import matter from "gray-matter";
import globby from "globby";
import fsx from "fs-extra";

import { onCreateNode, createPages } from "../gatsby-node";

// eslint-disable-next-line @typescript-eslint/no-var-requires
jest.mock("fs", () => require("memfs").fs);
jest.mock("gatsby-source-filesystem", () => {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const { basename } = require("path");
  return {
    createFilePath: ({ node }) =>
      `/${node.parent.relativeDirectory}/${basename(node.fileAbsolutePath)}/`,
  };
});

const ROOT = path.resolve(__dirname, "../src/documentation");

const getNode = n => n; // not really, but good enough

function getDirectoryNode(relativePath) {
  return {
    internal: { type: "Directory" },
    sourceInstanceName: "documentation",
    relativePath,
    absolutePath: `${ROOT}/${relativePath}`,
  };
}

function getMetaFileNode(relativePath) {
  return {
    internal: { type: "File" },
    sourceInstanceName: "documentation",
    relativePath,
    relativeDirectory: path.dirname(relativePath),
    absolutePath: `${ROOT}/${relativePath}`,
    base: path.basename("meta.yml"),
  };
}

function getMdxNode(relativePath, frontmatter) {
  return {
    internal: { type: "Mdx" },
    fileAbsolutePath: path.join(ROOT, relativePath),
    frontmatter,
    fields: {},
    parent: {
      internal: { type: "File" },
      sourceInstanceName: "documentation",
      relativeDirectory: path.dirname(relativePath),
    },
  };
}

const cacheMap = new Map();
const cache = {
  get: async key => cacheMap.get(key),
  set: async (key, value) => cacheMap.set(key, value),
};

async function createBreadcrumbs() {
  const files = globby.sync(ROOT);

  await Promise.all(
    files
      .filter(file => file.endsWith("meta.yml"))
      .map(async metaFile => {
        const node = getMetaFileNode(path.relative(ROOT, metaFile));
        // @ts-expect-error TODO
        await onCreateNode({ cache, node, getNode });
      }),
  );

  const breadcrumbs = await Promise.all(
    files
      .filter(file => !file.endsWith("meta.yml"))
      .sort()
      .map(async file => {
        const relativePath = path.relative(ROOT, file);
        const content = fsx.readFileSync(file);
        const node = getMdxNode(relativePath, matter(content).data);
        const actions = {
          createNodeField: ({ node: n, name, value }) => {
            // eslint-disable-next-line no-param-reassign
            n.fields[name] = value;
          },
        };
        // @ts-expect-error TODO
        await onCreateNode({ cache, node, getNode, actions });
        // @ts-expect-error TODO
        return { file: relativePath, breadcrumbs: node.fields.breadcrumbs };
      }),
  );

  return breadcrumbs;
}

afterEach(() => {
  cacheMap.clear();
  vol.reset();
});

describe("gatsby-node", () => {
  describe("registering documents", () => {
    it("should report missing meta.yml files", async () => {
      const node = getDirectoryNode("01-getting-started");
      const reporter = { panicOnBuild: jest.fn() };
      // @ts-expect-error TODO
      await onCreateNode({ cache, node, reporter });
      expect(reporter.panicOnBuild.mock.calls).toMatchInlineSnapshot(`
        [
          [
            "Expected meta.yml file to exist in "docs/src/documentation/01-getting-started", every directory in "src/documentation" should have one",
          ],
        ]
      `);
    });

    it("should report missing fields in meta.yml", async () => {
      vol.fromJSON(
        {
          "./01-getting-started/meta.yml": dedent`
            foo: bar
          `,
        },
        ROOT,
      );
      const node = getMetaFileNode("01-getting-started/meta.yml");
      const reporter = { panicOnBuild: jest.fn() };
      // @ts-expect-error TODO
      await onCreateNode({ cache, node, reporter });
      // first time for "title", second for "type"
      expect(reporter.panicOnBuild.mock.calls).toMatchInlineSnapshot(`
        [
          [
            "Expected "docs/src/documentation/01-getting-started/meta.yml" to contain fields: title, type",
          ],
        ]
      `);
    });

    it("should report incorrect type value in meta.yml", async () => {
      vol.fromJSON(
        {
          "./01-getting-started/meta.yml": dedent`
            title: Getting started
            type: unknown
          `,
        },
        ROOT,
      );
      const node = getMetaFileNode("01-getting-started/meta.yml");
      const reporter = { panicOnBuild: jest.fn() };
      // @ts-expect-error TODO
      await onCreateNode({ cache, node, reporter });
      // first time for "title", second for "type"
      expect(reporter.panicOnBuild.mock.calls).toMatchInlineSnapshot(`
        [
          [
            "Expected the value of "type" in "docs/src/documentation/01-getting-started/meta.yml" to be one of: folder, tabs",
          ],
        ]
      `);
    });

    it("should create breadcrumb nodes", async () => {
      vol.fromJSON(
        {
          "./01-getting-started/meta.yml": dedent`
            title: Getting started
            type: folder
          `,
          "./01-getting-started/01-for-designers/meta.yml": dedent`
            title: For designers
            type: tabs
          `,
          "./01-getting-started/01-for-designers/01-kiwi.mdx": dedent`
            ---
            title: For Kiwi.com employees
            ---
          `,
          "./01-getting-started/02-for-developers.mdx": dedent`
            ---
            title: For developers
            description: Our components are served as an npm package.
            ---
          `,
        },
        ROOT,
      );

      expect(await createBreadcrumbs()).toMatchSnapshot();
    });
  });

  describe("overview pages", () => {
    it("should create overview pages", async () => {
      const createPage = jest.fn();

      vol.fromJSON(
        {
          "./01-getting-started/meta.yml": dedent`
            title: Getting started
            type: folder
          `,
          "./01-getting-started/01-for-designers/meta.yml": dedent`
            title: For designers
            type: tabs
          `,
          "./01-getting-started/01-for-designers/01-kiwi.mdx": dedent`
            ---
            title: For Kiwi.com employees
            ---
          `,
          "./01-getting-started/02-for-developers.mdx": dedent`
            ---
            title: For developers
            description: Our components are served as an npm package.
            ---
          `,
          "./03-components/meta.yml": dedent`
            title: Components
            type: folder
          `,
          "./03-components/04-overlay/meta.yml": dedent`
            title: Overlay
            type: folder
          `,
          "./03-components/04-overlay/dialog/meta.yml": dedent`
            title: Dialog
            description: Prompts users to take or complete an action.
            type: tabs
          `,
          "./03-components/04-overlay/dialog/01-guidelines.mdx": dedent`
            ---
            title: Guidelines
            ---
          `,
        },
        ROOT,
      );

      await createBreadcrumbs();
      // @ts-expect-error TODO
      await createPages({
        graphql: () =>
          Promise.resolve({
            data: {
              allMdx: {
                nodes: [],
              },
            },
          }),
        actions: { createPage },
        cache,
      });

      expect(createPage.mock.calls).toMatchInlineSnapshot(`
        [
          [
            {
              "component": "docs/src/templates/Overview.tsx",
              "context": {
                "breadcrumbs": [
                  {
                    "name": "Getting started",
                    "url": "/getting-started/",
                  },
                ],
                "description": undefined,
                "idx": 1,
                "pages": [
                  {
                    "description": "Our components are served as an npm package.",
                    "idx": 1,
                    "slug": "/getting-started/for-developers/",
                    "title": "For developers",
                  },
                  {
                    "description": undefined,
                    "hasReactTab": false,
                    "idx": 1,
                    "slug": "/getting-started/for-designers/",
                    "title": "For designers",
                  },
                ],
                "slug": "/getting-started/",
                "title": "Getting started",
              },
              "path": "/getting-started/",
            },
          ],
          [
            {
              "component": "docs/src/templates/Overview.tsx",
              "context": {
                "breadcrumbs": [
                  {
                    "name": "Components",
                    "url": "/components/",
                  },
                ],
                "description": undefined,
                "idx": 3,
                "pages": [
                  {
                    "description": undefined,
                    "idx": 4,
                    "slug": "/components/overlay/",
                    "title": "Overlay",
                  },
                ],
                "slug": "/components/",
                "title": "Components",
              },
              "path": "/components/",
            },
          ],
          [
            {
              "component": "docs/src/templates/Overview.tsx",
              "context": {
                "breadcrumbs": [
                  {
                    "name": "Components",
                    "url": "/components/",
                  },
                  {
                    "name": "Overlay",
                    "url": "/components/overlay/",
                  },
                ],
                "description": undefined,
                "idx": 4,
                "pages": [
                  {
                    "description": "Prompts users to take or complete an action.",
                    "hasReactTab": false,
                    "idx": 0,
                    "slug": "/components/overlay/dialog/",
                    "title": "Dialog",
                  },
                ],
                "slug": "/components/overlay/",
                "title": "Overlay",
              },
              "path": "/components/overlay/",
            },
          ],
        ]
      `);
    });
  });
});
