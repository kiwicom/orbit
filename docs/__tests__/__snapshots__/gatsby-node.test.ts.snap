// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-node registering documents should create breadcrumb nodes 1`] = `
Array [
  Object {
    "breadcrumbs": Array [
      Object {
        "name": "Getting started",
        "url": "/getting-started/",
      },
      Object {
        "name": "For designers",
        "url": "/getting-started/for-designers/",
      },
    ],
    "file": "01-getting-started/01-for-designers/01-kiwi.mdx",
  },
  Object {
    "breadcrumbs": Array [
      Object {
        "name": "Getting started",
        "url": "/getting-started/",
      },
      Object {
        "name": "For developers",
        "url": "/getting-started/for-developers/",
      },
    ],
    "file": "01-getting-started/02-for-developers.mdx",
  },
]
`;
